nomad/server.go

```diff
@@ -1653,6 +1659,42 @@ func (s *Server) setupSerf(conf *serf.Config, ch chan serf.Event, path string) (
 	return serf.Create(conf)
 }
 
+func (s *Server) maybeAutoBootstrapACLs() error {
+	if !s.config.ACLEnabled || s.config.ACLBootstrapToken == "" {
+		// Exit without bootstrapping the ACL system
+		return nil
+	}
+
+	resp := structs.ACLTokenUpsertResponse{}
+	err := s.RPC("ACL.Bootstrap",
+		&structs.ACLTokenBootstrapRequest{BootstrapSecret: s.config.ACLBootstrapToken},
+		&resp,
+	)
+
+	if err != nil {
+		// Already bootstrapped
+		if strings.Contains(err.Error(), "ACL bootstrap already done") {
+			s.logger.Warn("ACL subsystem already bootstrapped; you should remove the bootstrap_token value from your configuration files or -acl-bootstrap-token flag from your startup flags")
+		} else {
+			// Unexpected Error
+			s.logger.Warn("error in auto bootstrap; you must manually bootstrap ACLs", "err", err)
+		}
+		return nil
+	}
+
+	// Nonsense result
+	if len(resp.Tokens) != 1 {
+		return fmt.Errorf("received incorrect token count. expected 1; got %d", len(resp.Tokens))
```

Comment:
There's no prefix indicating what this token is for. Maybe something like:

```suggestion
		return fmt.Errorf("received incorrect token count during ACL bootstrap. expected 1; got %d", len(resp.Tokens))
```